### PR TITLE
Request, RequestDecode, swarm peers, stats bw

### PR DESCRIPTION
**Pull request by jbenet**
I needed a couple commands, so i implemented them:
- swarm peers
- stats bw

I wanted a general way to issue commands to be exported,
so i added that. As described in #92

### CODE REVIEW

File: shell.go [hunk 1]
```diff
@@ -831,3 +852,34 @@ func (s *Shell) ObjectStat(key string) (*ObjectStats, error) {
 
 	return stat, nil
 }
+
+// ObjectStat gets stats for the DAG object named by key. It returns
+// the stats of the requested Object or an error.
+func (s *Shell) StatsBW() (*p2pmetrics.Stats, error) {
+	v := &p2pmetrics.Stats{}
+	err := s.RequestDecode(context.TODO(), &v, "stats/bw")
```
Comment (magik6k) [inline shell.go]: Other functions use.`.Background`

File: shell.go [hunk 2]
```diff
@@ -92,6 +94,25 @@ func (s *Shell) newRequest(ctx context.Context, command string, args ...string)
 	return NewRequest(ctx, s.url, command, args...)
 }
 
+func (s *Shell) Request(ctx context.Context, command string, args ...string) (*Request, *Response, error) {
+	req := s.newRequest(ctx, command, args...)
```
Comment (magik6k) [inline shell.go]: Maybe allow passing option map here too?

File: shell.go (hunk 1 above)
Comment (Stebalien) [reply shell.go]: Even better, let's just take a context (it's a new command).

File: shell.go (hunk 2 above)
Comment (Stebalien) [reply shell.go]: good point

**Pull request merged**